aws/ec2_transit_gateway.go

```diff
@@ -502,6 +559,49 @@ func waitForEc2TransitGatewayRouteTableAssociationDeletion(conn *ec2.EC2, transi
 	return err
 }
 
+func waitForEc2TransitGatewayPeeringAttachmentCreation(conn *ec2.EC2, transitGatewayAttachmentID string) error {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{
+			ec2.TransitGatewayAttachmentStatePending,
+			"initiatingRequest",
```

Comment:
Should this be `initiating` according to the [EC2 API Reference](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_TransitGatewayPeeringAttachment.html)?

```suggestion
			ec2.TransitGatewayAttachmentStateInitiating,
```

Otherwise a comment should be left explaining the discrepancy. 👍

Reply:
@bflad comment added in b2e4d5d and [issue](https://github.com/aws/aws-sdk-go/issues/3284) opened.

when using `ec2.TransitGatewayAttachmentStateInitiating` which equates to `"initiating"` , the following error is given 
```
Error: error waiting for EC2 Transit Gateway Peering Attachment (tgw-attach-xxxxxxxxxxxxxxxxx) availability: unexpected state 'initiatingRequest', wanted target 'pendingAcceptance, available'. last error: %!s(<nil>)'
```

Reply:
Thank you so much for looking into this! 🕵️